extensions/src/objectDetection/index.ts

```diff
@@ -0,0 +1,154 @@
+  async init() {
+    this.enableVideo()
+    this.detector = await initializeObjectDetector();
+    this.imageHelper = getImageHelper(this.videoDimensions.width, this.videoDimensions.height);
+  }
+
+  /**
+   * Runs a loop for constantly updating the video frame with a new set of detections.
+   * Used for the continuous detection block.
+   */
+  private async detectionLoop() {
+    while (this.continuous) {
+      const frame = this.getVideoFrame("canvas");
+      const start = Date.now();
+      if (frame) {
+        const detections = await this.detector.detect(frame);
+        this.displayImageDetections(detections);
+      }
+      const elapsed = Date.now() - start;
+      await untilTimePassed(this.processFreq - elapsed);
+      this.clearFrame();
+    }
+  }
+
+  /**
+   * Clears the frame of all drawables (resets the frame to blank video).
+   */
+  clearFrame() {
+    while (this.drawables.length > 0) this.drawables.shift().destroy();
+  }
+
+  /**
+   * Displays the detections from the detector in the video frame.
+   * @param detections The detections object returned from the detector's detect method
+   */
+  async displayImageDetections(detections) {
+    const { drawables, imageHelper } = this;
+    const rects = imageHelper.createRects(detections.detections, this.color, this.thickness);
+    drawables.push(this.createDrawable(rects));
+  }
+
+  @block({
+    type: "command",
+    text: (delay) => `Set detection rate to ${delay}`,
+    arg: {
+      type: "number",
+      options: [60, 30, 10, 2, 1],
+      defaultValue: 10
+    }
+  })
+  setFrameRate(delay: number) {
+    this.processFreq = 1000 / delay;
+  }
+
+  @block({
+    type: "command",
+    text: (color) => `Set box color to ${color}`,
+    arg: "color"
+  })
+  setColor(color: RGBObject) {
+    this.color = rgbToHex(color);
+  }
+
+  @block({
+    type: "command",
+    text: (thickness) => `Set box thickness to ${thickness}`,
+    arg: {
+      type: "number",
+      defaultValue: 5,
+      handler: (x) => {
+        return isNaN(x) ? 5 : Math.min(Math.max(0, x), 50);
+      }
+    }
+  })
+  setThickness(thickness: number) {
+
+    // temporary solution until handler fix
+    this.thickness = isNaN(thickness) ? 5 : Math.min(Math.max(0, thickness), 50);
+  }
+
+  @block({
+    type: BlockType.Command,
+    text: (freezeTime) => `Detect objects for ${freezeTime} seconds`,
+    arg: { type: ArgumentType.Number, defaultValue: 1 }
+  })
+  async detectObject(freezeTime: number) {
+    const frame = this.getVideoFrame('canvas');
+    const detections = await this.detector.detect(frame);
+    this.displayImageDetections(detections);
+    await untilTimePassed(freezeTime * 1000);
+    this.clearFrame();
+  }
+
+  @block({
+    type: BlockType.Command,
+    text: (state) => `Turn continuous detection ${state}`,
+    arg: { type: ArgumentType.Boolean, options: [{ text: 'on', value: true }, { text: 'off', value: false }] }
+  })
+  async continuouslyDetectObjects(state) {
```

Comment:
You'll need to specify that the `state` argument is a boolean for typescript to be happy. Is your editor not showing you errors? Here's what it looks like for me:
<img width="698" alt="Screen Shot 2023-05-29 at 4 23 05 PM" src="https://github.com/mitmedialab/prg-extension-boilerplate/assets/95306112/485231f4-7d9d-4232-85ba-5fae031980e2">